js/modules/pattern-fill.src.js

```diff
@@ -0,0 +1,320 @@
+		// Replace with old, but add default color
+		oldColor.pattern.color = this.color;
+		this.color = this.options.color = oldColor;
+	} else {
+		// We have a color, no need to do anything special
+		proceed.apply(this, Array.prototype.slice.call(arguments, 1));
+	}
+});
+
+
+/**
+ * Merge series color options to points
+ */
+wrap(H.Point.prototype, 'applyOptions', function (proceed) {
+	var point = proceed.apply(this, Array.prototype.slice.call(arguments, 1)),
+		colorOptions = point.options.color;
+
+	// Only do this if we have defined a specific color on this point. Otherwise
+	// we will end up trying to re-add the series color for each point.
+	if (typeof colorOptions === 'object') {
+		// Move path definition to object, allows for merge with series path
+		// definition
+		if (
+			colorOptions.pattern &&
+			typeof colorOptions.pattern.path === 'string'
+		) {
+			colorOptions.pattern.path = {
+				d: colorOptions.pattern.path
+			};
+		}
+		// Merge with series options
+		point.color = point.options.color = merge(
+			point.series.options.color, colorOptions
+		);
+	}
+
+	return point;
+});
+
+
+/**
+ * Add functionality to SVG renderer to handle patterns as complex colors
+ */
+H.addEvent(H.SVGRenderer, 'complexColor', function (args) {
+	var color = args.args[0],
+		prop = args.args[1],
+		element = args.args[2],
+		pattern = color.pattern,
+		value = '#343434',
+		bBox;
+
+	// Skip and call default if there is no pattern
+	if (!pattern) {
+		return true;
+	}
+
+	// We have a pattern.
+	if (
+		pattern.image ||
+		typeof pattern.path === 'string' ||
+		pattern.path && pattern.path.d
+	) {
+		// Real pattern. Add it and set the color value to be a reference.
+
+		// If we don't have an explicit ID, compute a hash from the
+		// definition and use that as the ID. This ensures that points with
+		// the same pattern definition reuse existing pattern elements by
+		// default. We combine two hashes, the second with an additional
+		// preSeed algorithm, to minimize collision probability.
+		pattern.id = pattern.id || 'highcharts-pattern-' +
+			hashFromObject(pattern) + hashFromObject(pattern, true);
+
+		// Autocompute width/height for images
+		if (pattern.image && (!pattern.width || !pattern.height)) {
+			bBox = element.getBBox();
+			pattern.width = pattern.width || Math.round(bBox.width);
+			pattern.height = pattern.height || Math.round(bBox.height);
+		}
+
+		// Add it. This function does nothing if an element with this ID
+		// already exists.
+		this.addPattern(pattern);
+
+		value = 'url(' + this.url + '#' + pattern.id + ')';
+
+	} else {
+		// Not a full pattern definition, just add color
+		value = pattern.color || value;
+	}
+
+	// Set the fill/stroke prop on the element
+	element.setAttribute(prop, value);
+
+	// Allow the color to be concatenated into tooltips formatters etc. 
+	color.toString = function () {
+		return value;
+	};
+
+	// Skip default handler
+	return false;
+});
+
+
+/**
+ * Add the predefined patterns
```

Comment:
Do we have this in docs somewhere?

Reply:
I suggest we add it to the docs about colors once we decide to merge this in: https://www.highcharts.com/docs/chart-design-and-style/colors

Reply:
Sorry, shortcut. I was thinking about API and docs. Like we have [`chart.colors`](https://api.highcharts.com/highcharts/colors).

Reply:
Aha, the predefined patterns are not in the API currently. Maybe they should be, and should be part of themes. What do you think @TorsteinHonsi?

Reply:
Well, it is the same with `linearGradient` and `radialGradient`. I think what we're missing is a programmatic description of the `Color` type. See for example [Point.color](https://api.highcharts.com/class-reference/Highcharts.Point#.color). I think what we need to do is to add a `typedef` for color, then describe the different types.